api/v1alpha1/common_validation.go

```diff
@@ -34,37 +34,67 @@ const (
 	ValidateValueParseError = "parse value field error:%s"
 )
 
-// ValidateScheduler validates the scheduler and duration
-func ValidateScheduler(duration *string, scheduler *SchedulerSpec, spec *field.Path) field.ErrorList {
+// ValidateScheduler validates the InnerSchedulerObject
+func ValidateScheduler(schedulerObject InnerSchedulerObject, spec *field.Path) field.ErrorList {
+
 	allErrs := field.ErrorList{}
+
+	nextStart := schedulerObject.GetNextStart()
+	nextRecover := schedulerObject.GetNextRecover()
+	if nextStart.After(nextRecover) {
+		nextStartField := spec.Child("nextStart")
+		allErrs = append(allErrs, field.Invalid(nextStartField, nextStart,
+			fmt.Sprintf("the nextStart: \"%s\" can not after the nextRecover: \"%s\"", nextStart, nextRecover)))
+	}
+
 	schedulerField := spec.Child("scheduler")
 	durationField := spec.Child("duration")
+	duration, err := schedulerObject.GetDuration()
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(durationField, nil,
+			fmt.Sprintf("parse duration field error:%s", err)))
+	}
 
-	if duration != nil && scheduler != nil {
-		_, err := cron.ParseStandard(scheduler.Cron)
-		if err != nil {
-			allErrs = append(allErrs, field.Invalid(schedulerField.Child("cron"),
-				scheduler.Cron,
-				fmt.Sprintf("parse cron field error:%s", err)))
-		}
+	scheduler := schedulerObject.GetScheduler()
 
-		_, err = time.ParseDuration(*duration)
-		if err != nil {
-			allErrs = append(allErrs, field.Invalid(durationField,
-				*duration,
-				fmt.Sprintf("parse duration field error:%s", err)))
+	if duration != nil && scheduler != nil {
+		errs := validateSchedulerParams(duration, durationField, scheduler, schedulerField)
+		if len(errs) != 0 {
+			allErrs = append(allErrs, errs...)
 		}
+	} else if (duration == nil && scheduler != nil) || (duration != nil && scheduler == nil) {
+		allErrs = append(allErrs, field.Invalid(schedulerField, scheduler, ValidateSchedulerError))
+	}
+	return allErrs
+}
 
-		if len(allErrs) > 0 {
-			return allErrs
+func validateSchedulerParams(duration *time.Duration, durationField *field.Path, spec *SchedulerSpec, schedulerField *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if duration != nil && spec != nil {
+		cronField := schedulerField.Child("cron")
+		allErrs = append(validateCron(spec.Cron, cronField))
+
+		scheduler, _ := cronv3.ParseStandard(spec.Cron)
```

Comment:
why ignore this error?

Reply:
`validateCron` had validated the `cron`.

If ParseStandard  causes error,  the `scheduler` will be `nil`

So I think we can  ignore this error.

Reply:
I don't suggest ignoring error here. The logic could be changed in future and we may miss the root cause when error happens.

Reply:
I've already fixed it.  PTAL
@cwen0 @Yisaer